### ISSUE TITLE
App crashes again on low storage within SQLite

### ISSUE THREAD

**Issue by Moncader** [plaidev/karte-android-sdk#8]
We have crash reports coming in in our app on Sentry/Crashlytics when there is no storage left on the device.
The stack trace is from Karte.

## Expected Behavior
The app does not crash. For this library, probably should just silently fail.

## Current Behavior
The app crashes.

## Detailed Description
Stack trace:

```
SQLiteFullException
database or disk is full (code 13 SQLITE_FULL)
```

```
android.database.sqlite.SQLiteConnection in nativeExecuteForChangedRowCount
android.database.sqlite.SQLiteConnection in executeForChangedRowCount at line 832
android.database.sqlite.SQLiteSession in executeForChangedRowCount at line 756
android.database.sqlite.SQLiteStatement in executeUpdateDelete at line 66
android.database.sqlite.SQLiteDatabase in updateWithOnConflict at line 1722
android.database.sqlite.SQLiteDatabase in update at line 1668
io.karte.android.utilities.datastore.DataStore$Companion in update at line 85
io.karte.android.tracking.queue.Dispatcher in queueRetry at line 59
io.karte.android.tracking.queue.Dispatcher in request at line 376
io.karte.android.tracking.queue.Dispatcher in dequeue at line 340
io.karte.android.tracking.queue.Dispatcher in access$dequeue
io.karte.android.tracking.queue.Dispatcher$push$2 in invoke at line 4
io.karte.android.tracking.queue.Dispatcher$push$2 in invoke
io.karte.android.tracking.queue.DispatcherKt$sam$java_lang_Runnable$0 in run at line 2
android.os.Handler in handleCallback at line 883
android.os.Handler in dispatchMessage at line 100
android.os.Looper in loop at line 214
android.os.HandlerThread in run at line 67
```

**Comment by tikidunpon** [comment]
@Moncader 

Sorry for the delay in replying.

Could you fill in the following information for investigating the problem?

And if you are using a bridge SDK (such as Unity or React Native), please provide the details as well.

## SDK Version

- core x.x.x
- inappmessaging x.x.x
- notifications x.x.x
- variables x.x.x
- visualtracking x.x.x
- Karte Gradle Plugin x.x.x
## Android Version

Please indicate the Android version that reproduces the problem, such as Android 5.0 or higher

## Build Environment

- Android Studio Version x.x.x
- Gradle Version x.x.x
- Android Gradle Plugin Version x.x.x

**Comment by tikidunpon** [comment]
@Moncader 

If you are using an older SDK, please update to the latest version and see if this fixes the problem.

Here is the release note.
https://developers.karte.io/docs/release-notes-android-sdk-v2

**Comment by Moncader** [comment]
Hello @tikidunpon 

We're using the Flutter bridge SDK.
```
karte_core: 0.1.0
karte_in_app_messaging: 0.1.0
karte_variables: 0.1.0
```

As for which version of the Android SDK is being used, you don't specify a version in your gradle files for the Flutter builds so we don't specify the version and don't know exactly what was built.

The build date was March 11, 2021, so whatever version was newest at that point.

> Please indicate the Android version that reproduces the problem, such as Android 5.0 or higher

Though we can see this crash on all versions of Android from 5.0 up to 11, most seems to centred around 7.0 and 8.0 (though that may not mean anything since many of our users are 7.0).

**Comment by tikidunpon** [comment]
@Moncader

Thank you for providing the details, we need your project id and user id that reproduce this problem.
So, I will continue to communicate with you through private channels, and I close this issue.

**Comment by tikidunpon** [comment]
@Moncader

We have released core 2.11.0, that fixes this issue, so please update your dependencies.
https://developers.karte.io/docs/release-notes-android-sdk-v2#core-2111

Thank you for reporting.
